src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -555,6 +553,251 @@ static void ucc_tl_ucp_ctx_remote_pack_data(ucc_tl_ucp_context_t *ctx,
     }
 }
 
+ucc_status_t ucc_tl_ucp_mem_map_memhbuf(ucc_tl_ucp_context_t *ctx,
+                                        void *pack_buffer, ucp_mem_h *mh)
+{
+    ucp_mem_map_params_t mmap_params;
+    ucs_status_t         status;
+    void                *packed_memh;
+
+    *mh = NULL;
+    /* unpack here */
+    packed_memh = UCC_TL_UCP_MEMH_TL_PACKED_MEMH(pack_buffer);
+
+    mmap_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_EXPORTED_MEMH_BUFFER;
+    mmap_params.exported_memh_buffer = packed_memh;
+
+    status = ucp_mem_map(ctx->worker.ucp_context, &mmap_params, mh);
+    if (UCS_OK != status) {
+        if (status != UCS_ERR_UNREACHABLE) {
```

Comment:
what does `UCS_ERR_UNREACHABLE` means in this context? I'm failing to understand from the doc.  Can you help me understand why should we return `UCC_OK` but "warn" in this case?

Reply:
Thanks for catching this. This was old code when there wasn't a clear divide between import on a host vs on offload device. `UCS_ERR_UNREACHABLE` is an error code in UCP when performing a memory map from an exported memory handle where the importing process cannot reach the memory domains of the exported process. For example, two hosts (host1 and host2) with two DPUs (bf1 and bf2) and bf2 tries to import a memory handle from host1. In this case, it was occurring on separate hosts that did not need to import the memory handle.